packages/shared/src/components/modals/SteppedModal.tsx

```diff
@@ -3,37 +3,86 @@ import React, {
   KeyboardEvent,
   MouseEvent,
   ReactElement,
+  useContext,
   useState,
 } from 'react';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import useAuthForms from '../../hooks/useAuthForms';
+import { AuthEventNames } from '../../lib/auth';
+import { logout } from '../../lib/user';
+import AuthOptions, { AuthDisplay } from '../auth/AuthOptions';
+import { DiscardAuthModal } from '../auth/common';
 import { Button } from '../buttons/Button';
 import TabContainer, { Tab } from '../tabs/TabContainer';
+import { SimpleTooltip } from '../tooltips/SimpleTooltip';
 import { ModalProps, StyledModal } from './StyledModal';
 
+type ValidateFunction = () => boolean | Promise<boolean>;
+
 interface SteppedModalProps extends ModalProps {
+  trigger: string;
+  skippable?: boolean;
+  invalidMessage?: string;
   isLastStepLogin?: boolean;
+  isFirstStepIntroduction?: boolean;
+  onValidateNext?: ValidateFunction[];
   onStepChange?: (stepBefore: number, stepNow: number) => void | Promise<void>;
+  onInvalid?: (step: number) => void | Promise<void>;
   onFinish?: () => void | Promise<void>;
+  onAuthSuccess?: (isLogin?: boolean) => void | Promise<void>;
 }
 
 function SteppedModal({
+  trigger,
   children,
+  skippable = true,
+  invalidMessage,
   isLastStepLogin,
+  isFirstStepIntroduction,
+  onValidateNext = [],
+  onInvalid,
   onStepChange,
   onFinish,
   onRequestClose,
+  onAuthSuccess,
   ...props
 }: SteppedModalProps): ReactElement {
   const [step, setStep] = useState(0);
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [screenValue, setScreenValue] = useState<AuthDisplay>(
+    AuthDisplay.Default,
+  );
+  const {
+    onDiscardAttempt,
+    onDiscardCanceled,
+    onContainerChange,
+    formRef,
+    container,
+    isDiscardOpen,
+  } = useAuthForms({
+    onDiscard: onRequestClose,
+  });
+
+  const onCancelAuth = async () => {
+    await logout();
+    await onStepChange?.(step, step - 1);
+    setStep(step - 1);
+    trackEvent({
+      event_name: AuthEventNames.CloseSignUp,
+      extra: JSON.stringify({ trigger, screenValue }),
+    });
+  };
```

Comment:
Doesn't the auth modal take care of it already?

Reply:
Just looked it up and yep, this should not be needed anymore 🚢

Reply:
My bad. After checking again, it seems the code block for tracking the close event was available on `AuthModal`. So it is necessary for us to have this here.